### ISSUE TITLE
rules-of-hooks false positive

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno_lint#1389]
### Lint Name

rules-of-hooks

### Code Snippet

```ts
import { useSignal } from "@preact/signals";

export function Comp() {
  const example = useSignal("example");
  return <div>{example}</div>;
}
```

### Expected Result

No lint errors.

### Actual Result

An `rule-of-hooks` error of:

```
Hook function called outside of a Component
Ensure that hooks are only called inside Components or custom hook functions
```

At the `useSignal("example")` in the example.

### Version

deno 2.1.7 (stable, release, aarch64-apple-darwin)
v8 13.0.245.12-rusty
typescript 5.6.2

(though this is a build off main)

**Comment by bartlomieju** [comment]
CC @marvinhagemeister

**Comment by marvinhagemeister** [comment]
Resolved by https://github.com/denoland/deno_lint/pull/1387

**Comment by kitsonk** [comment]
Excellent (I am loving the new lint rules though by the way, fixed up several things I wasn't even aware of!) 😍
